addons/l10n_latam_check/models/account_payment.py

```diff
@@ -0,0 +1,271 @@
+                    # check outbound payment and transfer or inbound transfer
+                    raise ValidationError(_(
+                        'Check "%s" is not anymore in journal "%s", it seems it has been moved by another payment.') % (
+                            rec.l10n_latam_check_id.display_name, rec.journal_id.name
+                            if rec.payment_type == 'outbound' else rec.destination_journal_id.name))
+                elif rec.payment_type == 'inbound' and not rec.is_internal_transfer and \
+                        rec.l10n_latam_check_id.l10n_latam_check_current_journal_id:
+                    raise ValidationError(_("Check '%s' is on journal '%s', we can't receive it again") % (
+                        rec.l10n_latam_check_id.display_name, rec.journal_id.name))
+
+        res = super().action_post()
+
+        # mark own checks that are not printed as sent
+        for rec in self.filtered(lambda x: x.check_number):
+            sequence = rec.l10n_latam_checkbook_id.sequence_id
+            sequence.sudo().write({'number_next_actual': int(rec.check_number) + 1})
+            rec.write({'is_move_sent': True})
+        return res
+
+    @api.onchange('payment_method_line_id', 'is_internal_transfer', 'journal_id', 'destination_journal_id')
+    def reset_check_ids(self):
+        """ If any of this fields changes the domain of the selectable checks could change """
+        self.l10n_latam_check_id = False
+
+    @api.onchange('check_number')
+    def _onchange_check_number(self):
+        for rec in self.filtered(lambda x: x.journal_id.company_id.country_id.code == "AR"):
+            try:
+                if rec.check_number:
+                    rec.check_number = '%08d' % int(rec.check_number)
+            except Exception:
+                pass
+
+    @api.depends('l10n_latam_check_operation_ids.state')
+    def _compute_l10n_latam_check_current_journal(self):
+        new_checks = self.filtered(lambda x: x.payment_method_line_id.code == 'new_third_checks')
+        for rec in new_checks:
+            last_operation = rec.env['account.payment'].search(
+                [('l10n_latam_check_id', '=', rec.id), ('state', '=', 'posted')], order="date desc, id desc", limit=1)
+            if not last_operation:
+                rec.l10n_latam_check_current_journal_id = rec.journal_id
+                continue
+            if last_operation.is_internal_transfer and last_operation.payment_type == 'outbound':
+                rec.l10n_latam_check_current_journal_id = last_operation.paired_internal_transfer_payment_id.journal_id
+            elif last_operation.payment_type == 'inbound':
+                rec.l10n_latam_check_current_journal_id = last_operation.journal_id
+            else:
+                rec.l10n_latam_check_current_journal_id = False
+
+    @api.model
+    def _get_trigger_fields_to_synchronize(self):
+        res = super()._get_trigger_fields_to_synchronize()
+        return res + ('check_number',)
```

Comment:
Change the Journal Items when the check number/name changes (including the check number there) -> for bank statement reconciliation -> to make it easier.

 @vbe-odoo research if this will be handy for US customers (the ones using the check_printing module) to see if we should add this in the main module and not only for the latam_check...) (so probably another PR will be better if included in the check_printing module

cc' @jjscarafia

Reply:
@jco-odoo @jjscarafia about this point, we had a few answers, and also talking with the Accounting App Expert in the SF office, they said that this is not really required but it will be a nice to have indeed (add some value), there is one client that asked for this and it will be in handy for him.

So, if it is too hard to include it in the check_printing module probably we can leave it outside, and for those special cases that want this, they could install the latam_check and use "Manual Numbering" and then they could have this extended feature.

Just brainstorming, we can discuss it further tomorrow.

Thanks!
cc' @ren-odoo @fvz-odoo

Reply:
for now we keep only on latam module